game-app/game-core/src/main/java/games/strategy/engine/framework/startup/ui/PlayerTypes.java

```diff
@@ -83,13 +45,7 @@ public PlayerTypes(final Collection<Type> playerTypes) {
   }
 
   public static Collection<Type> getBuiltInPlayerTypes() {
-    return List.of(
-        PlayerTypes.HUMAN_PLAYER,
-        PlayerTypes.WEAK_AI,
-        PlayerTypes.FAST_AI,
-        PlayerTypes.PRO_AI,
-        PlayerTypes.CLIENT_PLAYER,
-        PlayerTypes.BATTLE_CALC_DUMMY);
+    return List.of(PlayerTypes.WEAK_AI, PlayerTypes.FAST_AI, PlayerTypes.PRO_AI);
```

Comment:
Should this be called getBuildInAiPlayerTypes()?

Reply:
built-in is grammatically correct.

Reply:
Sorry for the typo, I was actually asking if "Ai" should be in the name, given this only returns ai types now.

Reply:
Ah my bad didn't notice it. Anyways I believe Fast_AI and Pro_AI can be removed from this list as well and moved to game-headed. There's really few code that is actually required for game-headless and therefore game-core

Reply:
I don't understand why AI should go to the headed codebase. Seems like they should be independent of UI.

I think right now there is a dependency in the ProLogUi code for appending to a swing text box, but I think only that part should be refactored out behind an interface.

I think it would be confusing if some Said were in one project and others in another.

Reply:
@asvitkine You have a point and we might need to introduce another submodule eventually (or just move the AI code to `:game-app:ai`?). Ideally `:game-app:game-core` will only contain code that is shared between `:game-app:game-headless` so that includes essential game logic. There is a point to be discussed: "Is AI essential game-logic?" The headless client only offers weak ai, so why would be binary need to be shipped with all the code necessary for the more sophisticated AI approaches?
In essence it's safe to say that while all UI code should be banned from `:game-app:game-core` this doesn't mean that all non-ui code should remain there.